### ISSUE TITLE
Set different rules per docker image in Artifact Registry repository

### ISSUE THREAD

**Issue by christidis** [mirakl/terraform-google-gcr-cleaner#52]
### TL;DR

Setting different rules per docker image in Container Registry (GCR) is well documented in the README file (with the python, nginx examples) and in the complete/minimal examples.

However, having the same functionality for docker registries in Artifact Registry (GAR) is not clear.
Copying the example (from the "complete" example)
```
  gar_repositories = [
    {
      project_id = "foobar-123"
      region     = "europe-west1"
      name       = "myrepo"
      parameters = {
        grace      = "180h"
        keep       = 3
        tag_filter = "^alpha.+$"
      }
    }
  ]
```

I have tried setting multiple dictionaries for each registry with different names like this
```
  gar_repositories = [
    {
      project_id = "foobar-123"
      region     = "europe-west1"
      name       = "myrepo/python"
      parameters = {
        grace      = "180h"
        keep       = 3
        tag_filter = "^alpha.+$"
      }
    },
    {
      project_id = "foobar-123"
      region     = "europe-west1"
      name       = "myrepo/nginx"
      parameters = {
        grace      = "10h"
        keep       = 2
      }
    }
  ]
```

This creates the jobs in Cloud Scheduler as expected but `terraform plan` will trigger a `forced replacement` on every execution: eg
```
  # module.gcr_cleaner.google_artifact_registry_repository_iam_member.this["docker-test/alpine"] must be replaced
-/+ resource "google_artifact_registry_repository_iam_member" "this" {
      ~ etag       = "BwXip8TxMiM=" -> (known after apply)
      ~ id         = "projects/test-artifact/locations/us-central1/repositories/docker-test/roles/artifactregistry.repoAdmin/serviceAccount:gcr-cleaner-sa-id@test-artifact.iam.gserviceaccount.com" -> (known after apply)
      ~ repository = "projects/test-artifact/locations/us-central1/repositories/docker-test" -> "projects/test-artifact/locations/us-central1/repositories/docker-test/alpine" # forces replacement
        # (4 unchanged attributes hidden)
    }
```

### Expected behavior

Setting full repository paths in the `gar_repositories` `name` field should not raise a `force recreation` with every terraform plan.

### Observed behavior

Setting full repository paths in the `gar_repositories` `name` field raises a terraform `force replacement` with every terraform plan.

### Terraform Configuration

```hcl
>cat gcp_cleaner.tf 
module "gcr_cleaner" {
  source  = "mirakl/gcr-cleaner/google"
  version = "1.3.0"
  # insert the 2 required variables here

  app_engine_application_location          = "us-central1"
  cloud_run_service_name                   = "gcr-cleaner-test"
  cloud_run_service_location               = "us-central1"
  cloud_run_service_maximum_instances      = 3
  cloud_run_service_timeout_seconds        = 300
  gcr_cleaner_image                        = "us-docker.pkg.dev/gcr-cleaner/gcr-cleaner/gcr-cleaner:latest"
  cloud_scheduler_job_schedule             = "*/1 * * * *"
  cloud_scheduler_job_time_zone            = "UTC"
  cloud_scheduler_job_attempt_deadline     = 600
  cloud_scheduler_job_retry_count          = 3
  cloud_scheduler_job_min_backoff_duration = 10
  cloud_scheduler_job_max_backoff_duration = 300
  cloud_scheduler_job_max_retry_duration   = 10
  cloud_scheduler_job_max_doublings        = 2

  gar_repositories = [
    {
      project_id = "test-artifact"
      region     = "us-central1"
      name       = "docker-test/nginx"
      dry_run    = "true"
      
    },
    {
      project_id = "test-artifact"
      region     = "us-central1"
      name       = "docker-test/alpine"
      dry_run    = "true"
    }
  ]
}
```

### Terraform Version

```sh
v1.0.11
```
